neurom/core/tests/test_soma.py

```diff
@@ -26,184 +26,175 @@
-    check_SomaC(SOMA_SIMPLECONTOUR_PTS_6)
-    check_SomaC(SOMA_SIMPLECONTOUR_PTS_8)
 
-
-@nt.raises(SomaError)
+def test_SomaC():
+    with warnings.catch_warnings(record=True):
+        check_SomaC(u'''((CellBody)
+                        (1 0 0 44)
+                        (0 1 0 44)
+                        (-1 0 0 44)
+                        (0 -1 0 44)) ''')
+
+        sin_pi_by_4 = math.cos(math.pi/4.)
+        cos_pi_by_4 = math.sin(math.pi/4.)
+
+        check_SomaC(u'''((CellBody)
+                         (1 0 0 44)
+                         ({sin} {cos} 0 44)
+                         (0 1 0 44)
+                         (-1 0 0 44)
+                         (-{sin} -{cos} 0 44)
+                         (0 -1 0 44))'''.format(sin=sin_pi_by_4,
+                                                cos=cos_pi_by_4))
+
+        check_SomaC(u'''((CellBody)
+        (1 0 0 44)
+        ({sin} {cos} 0 44)
+        (0 1 0 44)
+        (-{sin} {cos} 0 44)
+        (-1 0 0 44)
+        (-{sin} -{cos} 0 44)
+        (0 -1 0 44)
+        ({sin} -{cos} 0 44))'''.format(sin=sin_pi_by_4,
+                                       cos=cos_pi_by_4))
+
+
+@nt.raises(RawDataError)
 def test_invalid_soma_points_0_raises_SomaError():
-    _soma.make_soma(INVALID_PTS_0)
+    with warnings.catch_warnings(record=True):
+        load_neuron(StringIO(u'''((Dendrite)
+                                 (0 0 0 44)
+                                 (0 -44 0 44)
+                                 (0 +44 0 44))'''), reader='asc').soma
 
 
 @nt.raises(SomaError)
 def test_invalid_soma_points_2_raises_SomaError():
-    _soma.make_soma(INVALID_PTS_2)
+    with warnings.catch_warnings(record=True):
+        load_neuron(StringIO(u'''((CellBody)
+                                 (0 0 0 44)
+                                 (0 +44 0 44))'''), reader='asc').soma
 
 
-def test_make_Soma_Cylinders():
-    points = [[0, 0, -10, 40],
-              [0, 0,   0, 40],
-              [0, 0,  10, 40],
-              ]
-    s = _soma.SomaCylinders(points)
+def test_Soma_Cylinders():
+    s = load_neuron(StringIO(u'''
+                1 1 0 0 -10 40 -1
+                2 1 0 0   0 40  1
+                3 1 0 0  10 40  2'''), reader='swc').soma
+
     # if r = 2*h (ie: as in this case 10 - -10 == 20), then the
     # area of a cylinder (excluding end caps) is:
     # 2*pi*r*h == 4*pi*r^2 == area of a sphere of radius 20
     nt.eq_(s.radius, 20.0)
     nt.assert_almost_equal(s.area, 5026.548245743669)
-    nt.eq_(s.center, [0, 0, -10])
+    assert_array_equal(s.center, [0, 0, -10])
     nt.ok_('SomaCylinders' in str(s))
 
     # cylinder: h = 10, r = 20
-    soma_2pt_normal = np.array([
-        [0.0,   0.0,  0.0, 20.0, 1, 1, -1],
-        [0.0, -10.0,  0.0, 20.0, 1, 2,  1],
-    ])
-    s = _soma.make_soma(soma_2pt_normal, soma_class=_soma.SOMA_CYLINDER)
+    s = load_neuron(StringIO(u'''
+                1 1 0   0 0 20 -1
+                2 1 0 -10 0 20  1'''), reader='swc').soma
```

Comment:
I'm now wondering if those 2 point cylinder correspond to a valid configuration ?

Reply:
I don't know to be honest